traces/assembly/jr.txt

```diff
@@ -5,24 +5,15 @@ ADDI R2 R0 10
 HALT
 
 Expected stats:
-5 total instructions
-3 arithmetic ops
+4 total instructions
+2 arithmetic ops
 2 control flow ops
-7 total cycles with NO forwarding
-- Need to confirm this, assumes:
-  - JR has 2 stalls
-  - New address is known at end of JR EX stage
-  - HALT is fetched in same cycle as JR memory
-2 stalls total
-
-6 total cycles with forwarding
-- Need to confirm this, assumes:
-  - JR has 1 stall
-  - New address is known at end of JR EX stage
-  - HALT is fetched in same cycle as JR memory
-1 stall total
+10 total cycles with or without forwarding
+2-cycle branch penalty
+- I don't think any stalls are recorded, but we will flush 2 instructions
```

Comment:
For this one and the bz-taken, I had been thinking about it wrong with regard to forwarding. These should not be impacted by whether or not we're forwarding. I think I just needed to take a little break in order to think about it right 🫠 

That said, please call it out if this seems potentially incorrect. And if anyone else wants me to step through their instruction sequences to verify expected stats, just let me know.

Reply:
Oof, revisiting again, and these are impacted by forwarding for sure. Going to work on fine-tuning that along with the usage of `ADDI` and resulting output.